Source/NSKeyValueBinding.m

```diff
@@ -122,6 +122,40 @@ - (void) unbind: (NSString *)binding
 
 @end
 
+@implementation NSString (PrivatePathExtension)
+
+- (NSString *) stringByRemovingFirstKeyPath
```

Comment:
This method looks like overkill. Why not just look for the first dot and remove everything before including the dot? And if there is no dot just return the string itself.
On the other hand we should also be careful not to provide to many additional methods on standard classes. This could easily clash with application code that adds a method of the same name. In my view it would be best to avoid this method, if there isn't something similar on Apple.

Reply:
I've moved the method to NSOutlineView as a private method.   I'm going to re-implement it to make it simpler.

Reply:
I have removed the private method entirely. :)